cosmos/node.go

```diff
@@ -1,123 +1,84 @@
 		return nil, err
 	}
 
+	// generate first user
 	account, err := kb.GenerateAccount(ccfg.GenesisAccount.Name, ccfg.GenesisAccount.Mnemonic, ccfg.GenesisAccount.Password)
 	if err != nil {
 		return nil, err
 	}
 
-	// create app database and create an instance of the app
-	db, err := db.NewGoLevelDB("app", ccfg.Path)
-	if err != nil {
-		return nil, err
-	}
-
-	logger := logger.TendermintLogger()
-	app, cdc := app.NewNameServiceApp(logger, db)
-
 	// build a message to create validator
 	msg := newMsgCreateValidator(
 		sdktypes.ValAddress(account.GetAddress()),
 		ed25519.PubKeyEd25519(ccfg.ValidatorPubKey),
 		ccfg.GenesisAccount.Name,
 	)
-	logrus.WithField("msg", msg).Info("validator tx")
-
-	// sign the message
-	signedTx, err := signTransaction(
-		cdc,
-		kb,
-		msg,
-		ccfg.ChainID,
-		ccfg.GenesisAccount.Name,
-		ccfg.GenesisAccount.Password,
-	)
+	signedTx, err := NewTxBuilder(cdc, 0, 0, kb, ccfg.ChainID).
+		Create(msg, ccfg.GenesisAccount.Name, ccfg.GenesisAccount.Password)
 	if err != nil {
 		return nil, err
 	}
-	logrus.WithField("signedTx", signedTx).Info("signed tx")
 
 	// initialize app state with first validator
-	appState, err := createAppState(cdc, account.GetAddress(), signedTx)
+	appState, err := createAppState(app.DefaultGenesis(), cdc, account.GetAddress(), signedTx)
 	if err != nil {
 		return nil, err
 	}
 
-	return node.NewNode(cfg,
-		me,
+	nodeKey, err := p2p.LoadOrGenNodeKey(cfg.NodeKeyFile())
+	if err != nil {
+		return nil, err
+	}
+
+	// init node
+	node, err := node.NewNode(
+		cfg,
+		privval.LoadOrGenFilePV(cfg.PrivValidatorKeyFile(), cfg.PrivValidatorStateFile()),
 		nodeKey,
-		proxy.NewLocalClientCreator(app),
+		proxy.NewLocalClientCreator(app.BaseApp()),
 		genesisLoader(cdc, appState, ccfg.ChainID, ccfg.GenesisTime),
 		node.DefaultDBProvider,
 		node.DefaultMetricsProvider(cfg.Instrumentation),
-		logger,
+		logger.TendermintLogger(),
 	)
-}
-
-func signTransaction(cdc *codec.Codec, kb *Keybase, msg sdktypes.Msg, chainID, accountName, accountPassword string) (authtypes.StdTx, error) {
-	fees := authtypes.NewStdFee(flags.DefaultGasLimit, sdktypes.NewCoins())
-	gasPrices := sdktypes.DecCoins{}
-	stdTx := authtypes.NewStdTx([]sdktypes.Msg{msg}, fees, []authtypes.StdSignature{}, "")
-
-	txBldr := authtypes.NewTxBuilder(
-		authutils.GetTxEncoder(cdc),
-		0,
-		0,
-		flags.DefaultGasLimit,
-		flags.DefaultGasAdjustment,
-		true,
-		chainID,
-		"",
-		sdktypes.NewCoins(),
-		gasPrices,
-	).WithKeybase(kb)
+	if err != nil {
+		return nil, err
+	}
 
-	return txBldr.SignStdTx(accountName, accountPassword, stdTx, false)
+	return node, nil
 }
 
-func createAppState(cdc *codec.Codec, address sdktypes.AccAddress, signedStdTx authtypes.StdTx) (map[string]json.RawMessage, error) {
-	appState := app.ModuleBasics.DefaultGenesis()
-
+func createAppState(defaultGenesis map[string]json.RawMessage, cdc *codec.Codec, address sdktypes.AccAddress, signedStdTx authtypes.StdTx) (map[string]json.RawMessage, error) {
```

Comment:
```suggestion
func createAppState(defaultGenesisŚtate map[string]json.RawMessage, cdc *codec.Codec, address sdktypes.AccAddress, signedStdTx authtypes.StdTx) (map[string]json.RawMessage, error) {
```

Here we create app state and the funciton recives `defaultGenesisState` (aka `appState` `defaultAppState`). `defaultGenesis` would mean the whole genesis struct and it wouldn't be true in that case.

Also `func SetGenTxsInAppGenesisState(cdc *codec.Codec, appGenesisState map[string]json.RawMessage` expects appGenesisState, not appGenesis